lib/gambit/unstable/unstable.scm

```diff
@@ -0,0 +1,60 @@
+;;;============================================================================
+
+;;; File: "unstable.scm"
+
+;;; Copyright (c) 1994-2021 by Marc Feeley, All Rights Reserved.
+
+;;;============================================================================
+
+;;; Experimental procedures. May be changed or removed at any time.
+
+(##include "unstable#.scm")
+
+;;;----------------------------------------------------------------------------
+
+;; TODO: The following are copied from lib/_num#.scm
+
+(##define-macro (macro-inexact--1) -1.0)
+
+(##define-macro (macro-make-exact-10^n-table)
+
+  (define max-exact-power-of-10 22)
+  ;; (floor (inexact->exact (/ (log (expt 2 (macro-flonum-m-bits-plus-1)))
+  ;;                           (log 5))))
+
+  (let ((t (make-vector (+ max-exact-power-of-10 1))))
+
+    (let loop ((i max-exact-power-of-10))
+      (if (not (< i 0))
+          (begin
+            (vector-set! t i (exact->inexact (expt 10 i)))
+            (loop (- i 1)))))
+
+    `',(list->f64vector (vector->list t))))
+
+(define exact-10^n-table (macro-make-exact-10^n-table))
+
+(define-prim&proc (make-inexact-real (negative? boolean)
+                                     (mantissa nonnegative-exact-integer)
+                                     (exponent nonnegative-exact-integer))
```

Comment:
Sorry, the exponent should be simply "exact-integer".

Moreover, maybe the `negative?` parameter should come last and default to `#f`.  This would make even more sense if `mantissa` is allowed to be a negative integer (either in the first version of the API or somewhere down the road).

Also the `exponent` could default to `0`, so `(make-inexact-real 123)` would give `123.0`.  This would be a good motivation for accepting negative integers so that `(make-inexact-real -123)` would give `-123.0`.

Reply:
True. If both `mantissa` and `exponent` are allowed to be negative, and the separate `negative?` argument is dropped, that would be ideal IMHO.

Reply:
Let's revisit the issue of which arguments to make optional once we've agreed on the full range of available arguments. If we're turning this into a SRFI, an argument for the R7RS precision (`e`, `s`, `f`, `d`, `l`) should be added.

Reply:
You can't eliminate the `negative?` parameter because of `-0.0`.